src/CommandHandler.h

```diff
@@ -64,6 +65,7 @@ class CommandHandler {
   Master *MasterPtr = nullptr;
   std::vector<std::unique_ptr<FileWriterTask>> FileWriterTasks;
   friend class ::T_CommandHandler;
+  friend class ::CommandHandler_Testing;
```

Comment:
I think it would be much nicer to test `CommandHandler` by constructing it with a fake `Master`, tasks would be registered with the fake. You then wouldn't need the `FileWriterTasks` vector and supporting logic or these `friend`s.
https://arne-mertz.de/2015/08/unit-tests-are-not-friends/
If I haven't missed some reason why that wouldn't work then you could create an issue for it, or otherwise try it in this PR.